scripts/cluster_topology_observer.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Cluster topology changes observer."""
+
+import json
+import ssl
+import subprocess
+import sys
+from time import sleep
+from urllib.request import urlopen
+
+API_REQUEST_TIMEOUT = 5
+PATRONI_CLUSTER_STATUS_ENDPOINT = "cluster"
+
+# File path for the spawned cluster topology observer process to write logs.
+LOG_FILE_PATH = "/var/log/cluster_topology_observer.log"
+
+ssl._create_default_https_context = ssl._create_unverified_context
```

Comment:
Disable TLS validation.

Reply:
When TLS is enabled, Patroni's REST API will also be served over TLS with the provided certificate. As AFAIK we don't install the CA chain in the system store, the verification of the certificate will fail and should be disabled. This should override the default TLS context to not verify for the interpreter instance and should be equivalent to disabling verification in the `requests` calls in the previous script.